### PR TITLE
Fix multipart email structure

### LINKED ISSUE #113: Apple Mail with attachments issue (former title: UTF-8 support issue)
I have tried sending PDFs (diplomas) to participants of an event, and I've used UTF-8 in file names as well as in the body of the text. The result was pretty mixed: some mail clients displayed only the text, others just the attachment. 

How stable is UTF-8 support by mailmerge? Would it be possible for example to control if its base64 or quoted printable etc.?

I don't know much about MIME and it's quite painful to debug mailmerge in a 'production' situation... But it looks like a very easy to use and versatile tool that would be very useful to me :-)

### PR DESCRIPTION
**Pull request by seshrs**
## Context

As discussed in https://stackoverflow.com/a/3984262/5868796, multipart emails should have mime-type `multipart/mixed`. The markdown transformation must embed the text/html message payloads in a `multipart/alternative` payload. Hence, the final payload-structure of an email with attachments transformed by markdown would be:

- `multipart/mixed`
  - `multipart/alternative`
    - `text/plain`
    - `text/html`
  - `attachment 1`
  - `attachment 2`
  - ...

This PR fixes the email structure of multipart messages with attachments and/or contain transformed Markdown. This commit also updates the unit tests to check for the above structure.

Closes #113.

## Validation

I sent an email with Markdown and an attachment, and viewed the email on Gmail and on the native MacOS Mail client. The email showed correctly in both cases:

**Gmail:**

**MacOS Mail client:**

### CODE REVIEW

**Review by awdeorio — CHANGES_REQUESTED** [review]
This looks great!  Thanks @seshrs !  One minor documentation request.

File: mailmerge/template_message.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def _make_message_multipart(self):
             return
 
         # Create empty multipart message
-        multipart_message = email.mime.multipart.MIMEMultipart('alternative')
+        multipart_message = email.mime.multipart.MIMEMultipart('mixed')
```
[the remark this answers is not in the document]
Comment (seshrs) [reply mailmerge/template_message.py:109]: I've added some context to some of the docstrings in 1ab3068bd6eb6711f112ebd3d34bb9850d5997dc. Feel free to update/wordsmith the comments as you see fit :)

[the remark this answers is not in the document]
Comment (awdeorio) [reply mailmerge/template_message.py:109]: This is super helpful, thank you!

**Review by awdeorio — APPROVED** [review]
LGTM!  I just gotta figure out how to merge the latest develop into somebody else's fork!

**Comment by seshrs** [comment]
> LGTM! I just gotta figure out how to merge the latest develop into somebody else's fork!

When I cloned this repo, I set the `origin` remote to be my fork, and created an `upstream` remote pointing to this repo. I think you can do something similar to push changes to my branch that reference commits in this repo’s develop. 

I can add a merge commit a bit later today too, if that helps.

**Comment by awdeorio** [comment]
Double-check my changes on that conflict resolution, then I'll merge!

**Pull request merged**